independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/jaxrs/SseBroadcasterImpl.java

```diff
@@ -72,7 +72,9 @@ public synchronized void close() {
 
     synchronized void fireClose(SseEventSinkImpl sseEventSink) {
         for (Consumer<SseEventSink> listener : onCloseListeners) {
-            listener.accept(sseEventSink);
+            if (sinks.remove(sseEventSink)) {
```

Comment:
I don't get what do you mean, maybe I'm missing something. If we don't remove the sink (which is closing) from the broadcaster, that sink will continue to receive events even if it is closed, right?

Reply:
You have `n` listeners. On the first one, you remove the sink, then it's removed, and for all remaining listeners the sink is already removed so you don't notify them. That's a logic bug, no?

Reply:
Oh, now I see it, sorry! You're right, the removal should happen after all the listeners have been notified. I'll convert this pr to draft and work some more on it, since I'm already adding some tests.